rootston/cursor.c

```diff
@@ -148,14 +154,36 @@ static void roots_passthrough_cursor(struct roots_cursor *cursor,
 	}
 }
 
+static void roots_grabbed_cursor(struct roots_cursor *cursor,
+		uint32_t time) {
+	struct roots_seat *seat = cursor->seat;
+	struct roots_view *view = cursor->grabbed_view;
+	double sx = cursor->cursor->x - view->x;
+	double sy = cursor->cursor->y - view->y;
+	if (view && view->wlr_surface) {
+		wlr_seat_pointer_notify_enter(seat->seat, view->wlr_surface, sx, sy);
```

Comment:
The seat keeps track of the surface that last got enter ("pointer focus") and won't do anything if you try to send enter to the same surface twice. If you send enter to a different surface, it will automatically send leave to the last surface that got enter.

The alternative would be to explicitly guard it, but the guard already happens internally so we don't bother.

Reply:
I don't think this needs guarding in particular, but it feels wrong to me that there could be any way for another view to have received an enter event between cursor leaving the surface area and the move -- so at best this enter here is noop because of the interal guard, but I'm worried it could hide another unwanted behaviour we wouldn't notice with this.

Reply:
This is just the standard idiom for sending pointer motion when the surface you want to send motion to is known. Motion and enter were separated because they are separate in the protocol and we want the seat to do whatever the protocol can do (although I don't think we have a known use case for enter without motion).